torch/nn/functional.py

```diff
@@ -3077,3 +3077,195 @@ def _pad_circular(input, padding):
         input = torch.cat([input[:, :, :, :, -(padding[-5] + padding[-6]):-padding[-5]], input], dim=4)
 
     return input
+
+
+@weak_script
+def multi_head_attention_forward(
+        query, key, value, embed_dim_to_check, num_heads,
+        in_proj_weight, in_proj_bias, bias_k, bias_v, add_zero_attn,
+        head_dim, scaling, dropout_p, out_proj, training=True,
+        key_padding_mask=None, need_weights=True, attn_mask=None):
+    # type: (Tensor, Tensor, Tensor, int, int, Tensor, Tensor, Tensor, Tensor, bool, int, float, float, Tensor, bool, Optional[Tensor], bool, Optional[Tensor]) -> Tuple[Tensor, Tensor]
+    r"""
+    Args:
+        query, key, value: map a query and a set of key-value pairs to an output.
+            See "Attention Is All You Need" for more details.
+        embed_dim_to_check: total dimension of the model.
+        num_heads: parallel attention heads.
+        in_proj_weight, in_proj_bias: input projection weight and bias.
+        bias_k, bias_v: bias of the key and value sequences to be added at dim=0.
+        add_zero_attn: add a new batch of zeros to the key and
+                       value sequences at dim=1.
+        head_dim: dimension of attention heads.
```

Comment:
head_dim is equal to embed_dim/num_heads. So you need to pass only two of the three parameters.

Reply:
Done.